src/Mod/Part/Gui/Command.cpp

```diff
@@ -1329,7 +1329,38 @@ CmdPartExtrude::CmdPartExtrude()
 void CmdPartExtrude::activated(int iMsg)
 {
     Q_UNUSED(iMsg);
-    Gui::Control().showDialog(new PartGui::TaskExtrusion());
+    /** If the user selects objects prior to executing the Extrude command
+     *  we can use those objects rather than opening the dialog
+     */
+
+    bool neverSkipDialog = App::GetApplication().GetUserParameter().GetGroup("BaseApp/Preferences/Mod/Part")->GetBool("NeverSkipDialog", false);
+    if (!neverSkipDialog) {
+        App::Document* doc = App::GetApplication().getActiveDocument();
+        Base::PyGILStateLocker lock;
+        try {
+            PyObject* module = PyImport_ImportModule("BasicShapes.SkipGui");
+            if (!module) {
+                throw Py::Exception();
+            }
+            Py::Module skipgui(module, true);
+            Py::Boolean result(skipgui.callMemberFunction("makeExtrude"));
+            bool skip = result.as_bool();
+            if (!skip){
+                doc->openTransaction("Make Extrude");
+                Gui::Control().showDialog(new PartGui::TaskExtrusion());
+                doc->commitTransaction();
+            }
+        }
+        catch (Py::Exception&) {
+            doc->abortTransaction();

```

Comment:
The call of abortTransaction() here is quite problematic because you don't know where inside the try-block the exception was raised. 
You check for Py::Exception and most likely this will happen when running the function of the script, i.e. before you actually open the transaction.
The next thing is that you open a transaction before opening the dialog but after running the script. So, if the script was running successfully the new object won't be part of the transaction.

To address all the issues you should move the execution + transaction handling into a separate function or lambda function:
```
auto callFunction = [](const Py::Object& module, const std::string& function, App::Document* doc, const char* text) {
    try {
        doc->openTransaction(text);
        Py::Boolean result(skipgui.callMemberFunction(function));
        bool ok = result.as_bool();
        ok ? doc->commitTransaction() : doc->abortTransaction();
        return ok;
    }
    catch (const Py::Exception&) {
        doc->abortTransaction();
        throw; // re-throw the exception
    }
};
```